engine/src/execution_graph/logic_controllers/LogicalProject.cpp

```diff
@@ -132,6 +134,60 @@ std::unique_ptr<cudf::column> evaluate_string_functions(const cudf::table_view &
         computed_col = cudf::strings::replace(column, target, repl);
         break;
     }
+    case operator_type::BLZ_STR_LEFT:
+    {
+        assert(arg_tokens.size() == 2);
+        RAL_EXPECTS(!is_literal(arg_tokens[0]), "LEFT function not supported for string literals");
+
+        cudf::column_view column = table.column(get_index(arg_tokens[0]));
+        RAL_EXPECTS(is_type_string(column.type().id()), "LEFT argument must be a column of type string");
+
+        int32_t start = 0;
+        int32_t length = std::max(std::stoi(arg_tokens[1]), 0);
+        int32_t end = start + length;
+
+        computed_col = cudf::strings::slice_strings(
+            column,
+            cudf::numeric_scalar<int32_t>(start, true),
+            cudf::numeric_scalar<int32_t>(end, true)
+        );
+        break;
+    }
+    case operator_type::BLZ_STR_RIGHT:
+    {
+        assert(arg_tokens.size() == 2);
+        RAL_EXPECTS(!is_literal(arg_tokens[0]), "RIGHT function not supported for string literals");
+
+        cudf::column_view column = table.column(get_index(arg_tokens[0]));
+        RAL_EXPECTS(is_type_string(column.type().id()), "RIGHT argument must be a column of type string");
+
+        // Need the length of each string and the user provided offset
+        // to calculate the correct starting positions.
+        // Because this can be negative for some strings, set minimum as 0
+        int32_t offset = std::max(std::stoi(arg_tokens[1]), 0);
+        std::unique_ptr<cudf::column> char_counts = cudf::strings::count_characters(column);
+
+        auto starting_positions = cudf::binary_operation(
+            char_counts->view(),
+            cudf::make_column_from_scalar(cudf::numeric_scalar<int32_t>(offset), table.num_rows())->view(),
+            cudf::binary_operator::SUB,
+            cudf::data_type{cudf::type_id::INT32}
+        );
+
+        auto starting_positions_clamped = cudf::binary_operation(
+            starting_positions->view(),
+            cudf::make_column_from_scalar(cudf::numeric_scalar<int32_t>(0), table.num_rows())->view(),
+            cudf::binary_operator::NULL_MAX,
+            cudf::data_type{cudf::type_id::INT32}
+        );
```

Comment:
This is using using a lot of memory, but I'm not sure of a better way to do this (as I don't know libcudf that well).

As an example, for `RIGHT(col, 5)`, we need to make sure that we're taking the last 5 characters of every string. Since strings have different lengths, I first calculate the string lengths and then subtract 5 from all of them to get the "correct" starting point for each string. Then, since this value can be negative (e.g., a string of length 3), I take the MAX of that column and a column of all 0. Then, this gets passed as the `start` to `slice_strings`.

Would love some feedback from folks on whether this makes sense, and if there may be a better approach.

cc @davidwendt if you have any bandwidth

Reply:
You should be able to just use `cudf::strings::slice_strings()` with a negative start value to slice from the right. 
Try this:
```
computed_col = cudf::strings::slice_strings( column, -offset );
```

Reply:
Wow, perfect. For some reason when I tried this locally it wasn't working. I must have been making an unrelated mistake. 

Thanks David!